frontend/workflows/serverexperimentation/package.json

```
@@ -0,0 +1,30 @@
+{
+  "name": "@clutch-sh/serverexperimentation",
```

Could we possibly look at keeping it within the experimentation package?

---

Not sure whether we want to keep this as part of `experimentation` package - let me explain why. So the idea here is to make `experimentation` package a generic package for people to use when they want to implement their own type of experimentation package on top of the provided package. 

`serverexperimentation` is supposed to be one of packages that benefits from the abstractions that are provided by `experimentation` - if we combine `serverexperimentation` and `experimentation` packages it will become harder for us to ensure that experimentation stays a generic type of module. Since they are separated it's kind of a given that `experimentation` packages provides all of the hooks that are needed in order for any other package to benefit from functionality provided by `experimentation`.

I would like to know whether @wgallagher has any thoughts about the current organization of code/packages too.

---

I agree that it would be better to keep them separated. The goal of 'experimentation' is to provide a generic experimentation framework. 'serverexperimentation' is a specific instance of this.

---

Sounds good. One nit would be to use a hyphen here so it's a bit more readable.
```suggestion
  "name": "@clutch-sh/server-experimentation",
```

---

>Ideally the single entry point is the experimentation package which lists all experiments and gives users the ability to create experiments (maybe with a select for the type)

That's the goal but I think that dependency wise we will want to "inject" screens responsible for the creation of new tests into the `experimentation` package via `clutch.config.js`. In fact, this is how `server-experimentation` package does it now - see `links` part below (other packages will follow this pattern):

```javascript
"@clutch-sh/experimentation": {
    listExperiments: {
      trending: true,
      componentProps: {
        columns: ["identifier", "targets", "type", "description"],
        experimentTypes: {
          "type.googleapis.com/clutch.chaos.serverexperimentation.v1.TestConfig": {
            mapping: e => { return e },
            links: [
              {
                displayName: "Start Abort Experiment",
                path: "/serverexperimentation/startabort",
              },
              {
                displayName: "Start Latency Experiment",
                path: "/serverexperimentation/startlatency",
              },
            ],
          },
        },
      },
    },
  },
  "@clutch-sh/serverexperimentation": {
    startAbortExperiment: {},
    startLatencyExperiment: {},
  },
```

---

Renamed the module @dschaller (per your suggestion)

---

Yeah, I guess for now this works but going forward it might be better to embed these with react components instead of relying on a URL. The main thing I'm worried about is we are still flushing out what the end product will be and by publishing an external package we won't be able to remove it if we decide to take an approach that doesn't mesh with having this code be external.

---

Yea, we will need to figure this out - I know that it's not a great answer to your concerns (that I share too) but both of these packages are marked as beta ones so breaking changes - although not something that we want to introduce - are acceptable until these packages stabilize more.

Happy to chat more about how we could organize this to avoid using URL going forward *and* having the level of flexibility we need to have to support external packages (from outside of clutch repo) depending on the `experimentation` package at the same time.